Basket-Donation-Payments/Basket-Donation-Payments/Models/APIManager.m

```diff
@@ -103,13 +103,35 @@ + (void) newStripeCustomerIdWithName:(NSString*)fullName andEmail:(NSString*)ema
         }
         else {
             NSDictionary *dataDict =[NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
-            NSLog(@"Created new Stripe Customer");
             completion(nil, dataDict[@"id"]);
         }
     }];
     [task resume];
+}
 
-
++ (void)getSourceChargeIdWithPaymentIntent:(NSString*)paymentIntentId withBlock:(void (^)(NSError *, NSString *))completion {
+    NSString *backendURL = [APIManager getAPISecretKeysDict][@"Backend_Server_Url"];
+    NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"%@get-paymentintent-charge", backendURL]];
+    NSDictionary *json = @{
+        @"payment_intent": paymentIntentId,
+    };
+    NSData *body = [NSJSONSerialization dataWithJSONObject:json options:0 error:nil];
+    NSMutableURLRequest *request = [[NSURLRequest requestWithURL:url] mutableCopy];
```

Comment:
Same here, `[NSMutableURLRequest requestWithURL:url]` instead of `[[NSURLRequest requestWithURL:url] mutableCopy]`.